### PR TITLE
Fix restore on Supervisor Cluster

### PR DESCRIPTION
**Pull request by xing-yang**
This PR fixes the problem of restoring a workload on Supervisor Cluster.  Tested with a StatefulSet.

This fix depends on https://github.com/vmware-tanzu/velero-plugin-for-vsphere/pull/191 so it should be merged after that one is merged.

### CODE REVIEW

**Comment by svcbot-qecnsdp** [comment]
Sorry, some of the pre-check tests are failing for this PR ! 

Please check the Jenkins job for more details
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/226/

**Comment by svcbot-qecnsdp** [comment]
Sorry, some of the pre-check tests are failing for this PR ! 

Please check the Jenkins job for more details
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/227/

**Comment by svcbot-qecnsdp** [comment]
Sorry, some of the pre-check tests are failing for this PR ! 

Please check the Jenkins job for more details
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/228/

File: pkg/plugin/restore_pvc_action_plugin.go [hunk 1]
```diff
@@ -144,3 +171,45 @@ func (p *NewPVCRestoreItemAction) Execute(input *velero.RestoreItemActionExecute
...
 }
 
+func (p *NewPVCRestoreItemAction) createPod(item runtime.Unstructured) (*velero.RestoreItemActionExecuteOutput, error) {
+	var pod corev1.Pod
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(item.UnstructuredContent(), &pod); err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	if metav1.HasAnnotation(pod.ObjectMeta, "vmware-system-vm-uuid") {
+		newPod := &corev1.Pod{
```
Comment (dsu-igeek) [inline pkg/plugin/restore_pvc_action_plugin.go:181]: Won't this remove all of the annotations?  I think we should only remove the bad ones.  Users will probably complain if their annotations get removed on restore.

File: pkg/plugin/backup_pvc_action_plugin.go [hunk 2]
```diff
@@ -46,13 +46,27 @@ func (p *NewPVCBackupItemAction) Execute(item runtime.Unstructured, backup *vele
 		return item, nil, nil
 	}
 
+	blocked, crdName, err := utils.IsObjectBlocked(item)
+
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "Failed during IsObjectBlocked check")
+	}
+
+	if blocked {
```
Comment (dsu-igeek) [inline pkg/plugin/backup_pvc_action_plugin.go:55]: Would it be easier if we only include pods as a resource on restore?  I think they are all independent.  Then we wouldn't get called for backup or delete (I think)

File: pkg/plugin/restore_pvc_action_plugin.go (hunk 1 above)
Comment (xing-yang) [reply pkg/plugin/restore_pvc_action_plugin.go:190]: Fixed.

File: pkg/plugin/backup_pvc_action_plugin.go (hunk 2 above)
Comment (xing-yang) [reply pkg/plugin/backup_pvc_action_plugin.go:55]: Fixed.

**Comment by svcbot-qecnsdp** [comment]
Sorry, some of the pre-check tests are failing for this PR ! 

Please check the Jenkins job for more details
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/235/

**Comment by xing-yang** [comment]
Pre-check failed due to "no space left on device" error, but I have tested this fix on my vanilla setup as well.

**Comment by svcbot-qecnsdp** [comment]
Congratulations !! 

All the pre-check tests passed
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/237/

**Pull request merged**